src/codeu/chat/ClientMain.java

```diff
@@ -34,6 +29,61 @@
   private static Time lastLogBackup;
   private static final long BACKUP_RATE_IN_MS = 30000;
 
+  private static Chat chat;
+
+  private static void reloadOldInterests() throws IOException {
+    // Open the transaction log file for reading
+    FileReader fileReader = new FileReader("data/transaction_log.txt");
+    BufferedReader bufferedReader = new BufferedReader(fileReader);
+
+    // Read the header lines of each transaction log
+    String line = bufferedReader.readLine();
+
+    System.out.println("Loading interest system...");
+
+    while(line != null) {
+
+      // Instantiate a Tokenizer to parse through log's data
+      Tokenizer logInfo = new Tokenizer(line);
+
+      String commandType = logInfo.next();
+
+      if(commandType.equals("ADD-INTEREST-USER")){
+        Uuid owner = Uuid.parse(logInfo.next());
+        Uuid follow = Uuid.parse(logInfo.next());
+
+        chat.addUserInterest(owner, follow);
+      }
+      else if(commandType.equals("REMOVE-INTEREST-USER")){
+        Uuid owner = Uuid.parse(logInfo.next());
+        Uuid follow = Uuid.parse(logInfo.next());
+
+        chat.removeUserInterest(owner, follow);
+      }
+      else if(commandType.equals("ADD-INTEREST-CONVERSATION")){
+        Uuid owner = Uuid.parse(logInfo.next());
+        Uuid follow = Uuid.parse(logInfo.next());
+
+        chat.addConvoInterest(owner, follow);
+      }
+      else if(commandType.equals("REMOVE-INTEREST-CONVERSATION")){
+        Uuid owner = Uuid.parse(logInfo.next());
+        Uuid follow = Uuid.parse(logInfo.next());
+
+        chat.removeConvoInterest(owner, follow);
+      }
+
+      line = bufferedReader.readLine();
+    }
+
+    LOG.info("Successfully restored last logged interest system state.");
+
+    System.out.println("Successfully loaded interest system!");
+
+    fileReader.close();
```

Comment:
`fileReader` will get closed by closing `bufferedReader` so this line isn't necessary.